test/checksum.h

```diff
@@ -0,0 +1,21 @@
+//Header Guard
+#ifndef __CHECKSUM__
+#define __CHECKSUM__
+
+//Dependencies
+#include <sys/types.h>
```

Comment:
Do all of these types need to be included in header?  It seems like some of these could be included in `checksum.c` instead of in here.